src/convert.jl

```diff
@@ -248,15 +248,40 @@ Converts a hyperrectangular set to a zonotope.
 
 ### Input
 
-- `Zonotope`
-- `H` -- hyperrectangular set
+- `Zonotope` -- type, used for dispatch
+- `H`        -- hyperrectangular set
 
 ### Output
 
 A zonotope.
 """
-function convert(::Type{Zonotope}, H::AbstractHyperrectangle)
-    return Zonotope(center(H), Diagonal(radius_hyperrectangle(H)))
+function convert(::Type{Zonotope}, H::AbstractHyperrectangle{N}) where {N}
+    if isflat(H)
+        r = radius_hyperrectangle(H)
+        n = length(r)
+
+        nzgen = 0
+        Gnz = Vector{N}()
+        sizehint!(Gnz, n * n)
+        @inbounds for (i, ri) in enumerate(r)
+            if ri != zero(N)
+                col = zeros(N, n)
+                col[i] = ri
+                append!(Gnz, col)
+                nzgen += 1
+            end
+        end
+        G = reshape(Gnz, n, nzgen)
```

Comment:
It is probably more efficient to initialize `Gnz = zeros(N, n*n)` (or even `Gnz = zeros(N, n, n)`) and then only edit the entry that is not zero. But we can keep this code as well.

Reply:
it is not clear to me why that alternative would be more efficient; in general, push / append are fast. but yes, those are possible alternatives and we can possibly micro-optimize this as well. the change is nevertheless much faster than the version in master (See tests in the minkowski_sum issue https://github.com/JuliaReach/LazySets.jl/issues/1975#issuecomment-587000133)